src/check_lines.rs

```diff
@@ -90,8 +93,9 @@ pub fn check_lines(
         }
 
         // Remove any ignored checks.
-        // TODO(charlie): Only validate checks for the current line.
-        for (index, check) in checks.iter().enumerate() {
+        while let Some((index, check)) =
+            checks_iter.next_if(|(_index, check)| check.location.row() <= lineno + 1)
```

Comment:
Should this condition be merged with the one below? I.e:

```rust
checks_iter.next_if(|(_index, check)| check.location.row() == lineno + 1)
```

Reply:
Yeah I think you're right -- do you agree @andersk?

Reply:
Is there still a possibility of a check on line 0? If so, it would prevent that merged condition from ever succeeding.

Reply:
I don't think so, the rows are always 1-indexed.

Reply:
I’m thinking specifically of #237, before which there were errors like

```
test.py:0:0: E999 SyntaxError: unexpected EOF while parsing at line 3 column 7
```

This happened because `Location::default()` is 0:0. We still use `Location::default()` when reporting `IOError`. Is that definitely going to be the only case?

I guess we can add an assertion for this at the top of the loop.